### PR TITLE
added turbolinks option for pageview observer

### PR DESCRIPTION
**Pull request by jolim**

### CODE REVIEW

File: lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb [hunk 1]
```diff
@@ -6,9 +6,18 @@
 </script>
 <% end %>
 
-<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+<script>
+<% if options[:turbolinks] %>
+  document.addEventListener('turbolinks:load', function(event) {
+    var url = event.data.url;
+    dataLayer.push({'event':'pageView','virtualUrl': url});
+  });
```
Comment (bumi) [inline lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb]: sadly I don't know enough about the google tag manager... is google observing the dataLayer array after the page is loaded? is it enough to add entries to the array to get tracked by google?

**Comment by bumi** [comment]
thanks for the PR! looks good to me. 
Sadly I don't know enough about how the google tag manager works. See my comment (https://github.com/railslove/rack-tracker/pull/88#discussion_r130382810) do you have any hints on that?

**Comment by jolim** [comment]
Let me get back to you shortly. I am following a published model from the internet, but I have yet to validate it in practice. Perhaps I was too quick to send you the PR….

**Comment by bumi** [comment]
@jolim any update on this? let me know if you need some help.

**Comment by jolim** [comment]
It does seem to be working appropriately. According to my GTM consultant, the code tracks pageviews correctly when people use turbolinks. The major caveat is that I cannot confirm this first-hand because I don’t know anything about GTM and haven’t seen the difference in output.

 

Given that it is an option (which I will add docs for now), it should be safe to include. You can get additional feedback from others in practice if you are comfortable incorporating this as an experimental feature.

File: lib/rack/tracker/version.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module Rack
   class Tracker
-    VERSION = '1.3.1'
+    VERSION = '1.3.3'
```
Comment (DonSchado) [inline lib/rack/tracker/version.rb]: while I really appreciate your engagement, 
please let the decision _when_ and _how_ to pump version to the maintainers. ;)

**Comment by DonSchado** [comment]
would be ok for me! thanks

File: lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb (hunk 1 above)
Comment (agustinv) [reply lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb]: This add a data variable to your GTM that you will need to trigger on it changing and being present.

You can modify your GTM to do add trigger based on History Change (built in variable).  Link this trigger to GA PageView Tag in addition to your all pages trigger.

image

Comment (agustinv) [reply lib/rack/tracker/google_tag_manager/template/google_tag_manager_head.erb]: https://support.google.com/tagmanager/answer/7679322?hl=en#:~:text=Help%20CenterCommunity-,History%20change%20trigger,in%20single%20page%20web%20applications.

**Pull request merged**